src/profile/external-identities/OAuth2Link.tsx

```diff
@@ -0,0 +1,149 @@
+import { Clickable } from '@terra-ui-packages/components';
+import React from 'react';
+import { useState } from 'react';
+import { LazyClipboardButton } from 'src/components/ClipboardButton';
+import { ButtonPrimary } from 'src/components/common';
+import { icon } from 'src/components/icons';
+import { Ajax } from 'src/libs/ajax';
+import { EcmLinkAccountResponse } from 'src/libs/ajax/ExternalCredentials';
+import colors from 'src/libs/colors';
+import { withErrorReporting } from 'src/libs/error';
+import * as Nav from 'src/libs/nav';
+import { useCancellation, useOnMount } from 'src/libs/react-utils';
+import * as Utils from 'src/libs/utils';
+import { OAuth2Provider } from 'src/profile/external-identities/OAuth2Providers';
+import { SpacedSpinner } from 'src/profile/SpacedSpinner';
+
+const styles = {
+  clickableLink: {
+    display: 'inline',
+    color: colors.accent(),
+    cursor: 'pointer',
+    fontWeight: 500,
+  },
+  idLink: {
+    container: {
+      display: 'grid',
+      marginBottom: '0.6rem',
+      border: `1px solid ${colors.dark(0.55)}`,
+      borderRadius: 4,
+    },
+    linkContentTop: (hasBottom) => ({
+      display: 'grid',
+      rowGap: '0.6rem',
+      backgroundColor: colors.light(0.2),
+      padding: '1.2rem',
+      borderRadius: hasBottom ? '4px 4px 0 0' : 4,
+    }),
+    linkContentBottom: {
+      padding: '1.2rem',
+    },
+    linkName: {
+      fontSize: 18,
+      fontWeight: 700,
+      marginBottom: '0.6rem',
+      display: 'inline',
+    },
+    linkDetailLabel: {
+      fontWeight: 700,
+      marginBottom: '0.6rem',
+      marginRight: '1.2rem',
+    },
+  },
+};
+
+interface OAuth2LinkProps {
+  queryParams: { state?: string; code?: string };
+  provider: OAuth2Provider;
+}
+export const OAuth2Link = (props: OAuth2LinkProps) => {
+  const {
+    queryParams: { state, code },
+    provider,
+  } = props;
+
+  const signal = useCancellation();
+  const [accountInfo, setAccountInfo] = useState<EcmLinkAccountResponse>();
+  const [accountLoaded, setAccountLoaded] = useState<boolean>(false);
+
+  useOnMount(() => {
+    const loadAccount = withErrorReporting(`Error loading ${provider.name} account`, async () => {
+      setAccountInfo(await Ajax(signal).ExternalCredentials(provider).getAccountLinkStatus());
+    });
+    const linkAccount = withErrorReporting(`Error linking ${provider.name} account`, async (code, state) => {
+      setAccountInfo(await Ajax(signal).ExternalCredentials(provider).linkAccountWithAuthorizationCode(code, state));
+    });
+
+    if (Nav.getCurrentRoute().name === 'ecm-callback' && state && JSON.parse(atob(state)).provider === provider) {
```

Comment:
I think we would also want to check if `Nav.getCurrentRoute().name === 'oauth-callback'` (or oauth_callback if you change the route name)